EIPS/eip-7441.md

```diff
@@ -0,0 +1,101 @@
+
+The full specification of the proposed change can be found in [`/_features/whisk/beacon-chain.md`](https://github.com/ethereum/consensus-specs/blob/a39abe388bc2d1abd5b4fd62fd18aed497956b30/specs/_features/whisk/beacon-chain.md). In summary:
+
+- Update `BeaconState` with fields needed to track validator trackers, commitments, and the two rounds of candidate election.
+- Add `select_whisk_candidate_trackers` to compute the next vector of candidates from the validator set.
+- Add `select_whisk_proposer_trackers` to compute the next vector of proposers from current candidates.
+- Add `process_whisk_updates` to epoch processing logic.
+- Add `process_whisk_opening_proof` to validate block proposer has knowledge of this slot's elected tracker.
+- Modify `process_block_header` to not assert proposer election with `get_beacon_proposer_index`, instead assert valid opening proof.
+- Update `BeaconBlockBody` with fields to submit opening proof, shuffled trackers with proof, and tracker registration with proof.
+- Add `get_shuffle_indices` to compute pre-shuffle candidate selection
+- Add `process_shuffled_trackers` to submit shuffled candidate trackers.
+- Add `process_whisk` to block processing logic.
+- Modify `apply_deposit` to register an initial unique tracker and commitment without entropy.
+
+## Rationale
+
+### Fields per validator
+
+Whisk requires appending one tracker `(rG,krG)` and one commitment `kG` per validator. Both are updated at max once through a validator's lifetime.
+
+Trackers are registered with a randomized base `(rG,krG)` to make it harder for adversaries to track them through shuffling gates. It can become an issue if the set of honest shufflers is small.
+
+### Identity binding
+
+Each tracker must be bound to a validator's identity to prevent multiple parties to claim the same proposer slot. Otherwise, it would allow proposers to sell their proposer slot, and cause fork-choice issues if two competing blocks appear.
+
+Whisk does identity binding by storing a commitment to the tracker's secret `kG` in the validator record. Storing the commitment also ensures the uniqueness of `k`.
+
+Alternatively, identity binding can be achieved by forcing the hash prefix of `hash(kG)` to match its validator index. However, validators would have to brute force `k` making bootstrap of the system harder for participants with fewer computational resources.
+
+Identity binding can also be achieved by setting `k = hash(nonce + pubkey)`. However, proposers will need to reveal `k` and be de-anonymized for repeated proposals on adjacent shuffling phases.
+
+### Alternative: non-single secret election
+
+Secret non-single leader election is based on protocol engineering rather than cryptography, thus much simpler and cheap than Whisk. However, it complicates the fork-choice and opens it up to potential MEV time-buying attacks, making it an unsuitable option at the time of writing.
```

Comment:
```suggestion
Secret non-single leader election could be based on protocol engineering rather than cryptography, thus much simpler and cheaper than Whisk. However, it complicates the fork-choice and opens it to potential MEV time-buying attacks, making it an unsuitable option at the time of writing.
```